File: pset1/cash.c
// Greedy Algorithm - finds the optimal solution for the problem
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float give_right_amount(string prompt);
int main(void)
{
    // For keeping track of coins 
    int coins_used;
    // Prompting user for amount of change
    float amount_in_dollars = give_right_amount("Change: ");
    int amount = round(amount_in_dollars * 100); // Converting $ into cents
    while (amount >= 25)
    {     
        coins_used = amount / 25;
        amount = amount % 25;           
    }
    while (amount < 25 && amount >= 10)
    {       
        coins_used = coins_used + amount / 10;
        amount = amount % 10;
    } 
    while (amount < 10 && amount >= 5) 
    {
        coins_used = coins_used + amount / 5;
        amount = amount % 5;
    } 
    if (amount < 5)
    {
        coins_used = coins_used + amount;      
    }
    printf("%i\n", coins_used);
}

float give_right_amount(string prompt)
{
    float amount_in_dollars;
    do
    {
        amount_in_dollars = get_float("%s", prompt);
    }
    while (amount_in_dollars <= 0);
    return amount_in_dollars;        
}

File: pset6/cash/cash.py
# Greedy Algorithm - finds the optimal solution for the problem
# prompt user for an amount of change
# always use the largest coin possible (use algorithm)
# keep track of coins used
# print the final number of coins print(coins)

from cs50 import get_float


def main():
    amount = get_positive_float("Change owed: ")
    amount = round(amount*100)
    coins_used = 0
    if (amount >= 25):
        coins_used = amount // 25
        amount = amount % 25
        print("amount = ", amount)
        print("coins = ", coins_used)
    if (amount < 25 and amount >= 10):
        coins_used += amount // 10
        amount = amount % 10
        print("amount = ", amount)
        print("coins = ", coins_used)
    if (amount < 10 and amount >= 5):
        amount = amount % 5
        coins_used += 1
    if (amount < 5):
        coins_used += amount
    print(coins_used)


def get_positive_float(prompt):
    while True:
        n = get_float(prompt)
        if n > 0:
            break
    return n


if __name__ == "__main__":
    main()